src/unittests/loaders_and_trees.spec.ts

```diff
@@ -0,0 +1,338 @@
+g.test('suite').fn(async t => {
+  t.shouldReject('Error', t.load('suite1'));
+  t.shouldReject('Error', t.load('suite1:'));
+});
+
+g.test('group').fn(async t => {
+  t.expect((await t.load('suite1:*')).length === 8);
+  t.expect((await t.load('suite1:foo,*')).length === 3); // x:foo,* matches x:foo:
+  t.expect((await t.load('suite1:bar,*')).length === 1);
+  t.expect((await t.load('suite1:bar,buzz,buzz,*')).length === 1);
+
+  t.shouldReject('Error', t.load('suite1:f*'));
+});
+
+g.test('test').fn(async t => {
+  t.shouldReject('Error', t.load('suite1::'));
+  t.shouldReject('Error', t.load('suite1:bar:'));
+  t.shouldReject('Error', t.load('suite1:bar,:'));
+
+  t.shouldReject('Error', t.load('suite1::*'));
+  t.shouldReject('Error', t.load('suite1:bar,:*'));
+  t.shouldReject('Error', t.load('suite1:bar:*'));
+
+  t.expect((await t.load('suite1:foo:*')).length === 3);
+  t.expect((await t.load('suite1:bar,buzz,buzz:*')).length === 1);
+  t.expect((await t.load('suite1:baz:*')).length === 4);
+
+  t.expect((await t.load('suite2:foof:bluh,*')).length === 1);
+  t.expect((await t.load('suite2:foof:bluh,a,*')).length === 1);
+});
+
+g.test('case').fn(async t => {
+  t.shouldReject('Error', t.load('suite1:foo::'));
+  t.shouldReject('Error', t.load('suite1:bar:zed,:'));
+
+  t.shouldReject('Error', t.load('suite1:foo:h*'));
+
+  t.shouldReject('Error', t.load('suite1:foo::*'));
+  t.shouldReject('Error', t.load('suite1:baz::*'));
+  t.shouldReject('Error', t.load('suite1:baz:zed,:*'));
+
+  t.shouldReject('Error', t.load('suite1:baz:zed:'));
+  t.shouldReject('Error', t.load('suite1:baz:zed:a=1;b=2*'));
+  t.shouldReject('Error', t.load('suite1:baz:zed:a=1;b=2;'));
+  t.shouldReject('SyntaxError', t.load('suite1:baz:zed:a=1;b=2,')); // tries to parse '2,' as JSON
+  t.shouldReject('Error', t.load('suite1:baz:zed:a=1,b=2')); // '=' not allowed in value
+  t.shouldReject('Error', t.load('suite1:baz:zed:b=2*'));
+  t.shouldReject('Error', t.load('suite1:baz:zed:b=2;a=1;_c=0'));
+  t.shouldReject('Error', t.load('suite1:baz:zed:a=1,*'));
+
+  t.expect((await t.load('suite1:baz:zed:*')).length === 2);
+  t.expect((await t.load('suite1:baz:zed:a=1;*')).length === 2);
+  t.expect((await t.load('suite1:baz:zed:a=1;b=2')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:a=1;b=2;*')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:b=2;*')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:b=2;a=1')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:b=2;a=1;*')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:b=3;a=1')).length === 1);
+  t.expect((await t.load('suite1:baz:zed:a=1;b=3')).length === 1);
+  t.expect((await t.load('suite1:foo:hello:')).length === 1);
+});
+
+g.test('partial_test,makeQueryString').fn(async t => {
+  const s = new TestQuerySingleCase('suite1', ['baz'], ['zed'], { a: 1, b: 2 }).toString();
+  t.expect((await t.load(s)).length === 1);
+});
+
+g.test('partial_test,match').fn(async () => {});
```

Comment:
Should this test be empty?

Reply:
oops. I moved all the cases from this into other tests.